component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java

```diff
@@ -390,4 +362,48 @@ public void mqttPublishWithoutBrokerCredentials() {
         AssertJUnit.assertEquals(3, count);
         AssertJUnit.assertTrue(eventArrived);
     }
+
+    @Test
+    public void mqttPublishTest() {
+        log.info("Test for persist and restore state ");
+        PersistenceStore persistenceStore = new InMemoryPersistenceStore();
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.setPersistenceStore(persistenceStore);
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(
+                "define stream FooStream (symbol string, price float, volume long); " + "@info(name = 'query1') "
+                        + "@sink(type='mqtt', url= 'tcp://localhost:1883', "
+                        + "topic='mqtt_publish_event_persist',username='mqtt-user', "
+                        + "password='mqtt-password', clean.session='true', message.retain='false', "
+                        + "quality.of.service= '1', keep.alive= '60'," + "@map(type='xml'))"
+                        + "Define stream BarStream (symbol string, price float, volume long);"
+                        + "from FooStream select symbol, price, volume insert into BarStream;");
+        InputHandler fooStream = siddhiAppRuntime.getInputHandler("FooStream");
+        try {
+            this.mqttTestClient = new MqttTestClient("tcp://localhost:1883", "mqtt_publish_event_persist", 1);
+        } catch (ConnectionUnavailableException e) {
+            AssertJUnit.fail("Could not connect to broker.");
+        }
+        siddhiAppRuntime.start();
+        try {
+            fooStream.send(new Object[] { "WSO2", 55.6f, 100L });
+            fooStream.send(new Object[] { "IBM", 75.6f, 100L });
+            fooStream.send(new Object[] { "WSO2", 57.6f, 100L });
+            Thread.sleep(500);
+            siddhiAppRuntime.persist();
+            siddhiAppRuntime.shutdown();
+            fooStream = siddhiAppRuntime.getInputHandler("FooStream");
+            siddhiAppRuntime.start();
+            siddhiAppRuntime.restoreLastRevision();
+            fooStream.send(new Object[] { "IBM", 75.6f, 100L });
+            Thread.sleep(500);
+
+        } catch (InterruptedException e) {
+            AssertJUnit.fail("Thread sleep was  interrupted");
+        }
+        count = mqttTestClient.getCount();
+        eventArrived = mqttTestClient.getEventArrived();
+        AssertJUnit.assertEquals(4, count);
```

Comment:
Can we also assert for the event payload

Reply:
fixed in 51669b4